### PR TITLE
[FIX] hr_holidays: display correct remaining leaves for flexible hours

### PR DESCRIPTION
**Pull request by Bertrand2**
Steps to reproduce the issue:
- have an employee with an allocation with an end date
- remove the employee's calendar
- go on his dashboard
- his remaining leaves amount is 0

This commit implements a new method on the resource mixin to fetch a calendar for the record even though it might not have one. The default behaviour is to fall back on the company's calendar to ensure a value. This is overrided in the `hr_contract` module so that the calendar of the contract is prioritized.

task-3609738

### CODE REVIEW

**Review by robcoekaerts — COMMENTED** [review]
With this change, 'no calendar' on an employee basically means having the regular 40 hour default company calendar. Is that how we want it to behave?

**Comment by Bertrand2** [comment]
> With this change, 'no calendar' on an employee basically means having the regular 40 hour default company calendar. Is that how we want it to behave?

That's the case only when you don't have contract installed or if you don't have a current contract.
Basically, the use case is that we should be able to remove the employee calendar if he is working according to a planning, which may vary overtime. When looking for duration of presence/absence, we still have to refer to a calendar though; that's why we refer to either the employee's contract calendar or the company calendar in order to have some reference.

However, your comment makes me realize that this would still bring issues as with contracts installed, we only look into one calendar which is the current one. It should be the relevant one instead, though I'm also not sure what should be the behaviour if multiple calendar are covered by the period looked into.
This also doesn't work for models inheriting from the `resource.mixin` that aren't employees, so I need to move that method anyway.

**Comment by Bertrand2** [comment]
Changed. I decided to avoid to handle the case of multiple calendars if multiple contracts are on the period looked into, because it would involve modifying a whole lot more and even though I haven't investigated it I think it might not be doable in stable.

File: addons/resource/models/resource_mixin.py [hunk 1]
```diff
@@ -76,6 +76,10 @@ def copy_data(self, default=None):
         default['resource_calendar_id'] = resource.calendar_id.id
         return super().copy_data(default)
 
+    def _get_calendar(self, date_from=None, date_to=None):
```
Comment (sofiagvaladze) [inline addons/resource/models/resource_mixin.py:79]: what's a point of variables here? :eyes:

**Review by sofiagvaladze — CHANGES_REQUESTED** [review]
@Bertrand2 can you rebase the branch? it's not green

**Comment by sofiagvaladze** [comment]
robodoo r+

**Pull request closed without merging**